src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -187,39 +186,37 @@ private void readSurfaceTile(final int tileNum, final List<AbstractIlluminaPosit
                     headerBuffer.clear();
                 }
             }
+        } catch (final IOException ioe) {
+            throw new RuntimeIOException(ioe);
+        }
+    }
 
-            if (headerOnly) {
-                return;
-            }
-
-            int totalCycleCount = 0;
+    private void readTileData(int tileNum, List<AbstractIlluminaPositionFileReader.PositionInfo> locs) {
+        int totalCycleCount = 0;
 
-            if (cycleData[totalCycleCount].tileInfo == null) {
-                throw new PicardException("Could not find tile " + tileNum);
-            }
+        if (cycleData[totalCycleCount].tileInfo == null) {
+            throw new PicardException("Could not find tile " + tileNum);
+        }
 
-            for (final int outputLength : outputLengths) {
-                for (int cycle = 0; cycle < outputLength; cycle++) {
-                    final CycleData currentCycleData = cycleData[totalCycleCount];
-                    try {
-                        if (cachedTile[totalCycleCount] == null) {
-                            if (!cachedFilter.containsKey(cycleData[totalCycleCount].tileInfo.tileNum)) {
-                                cacheFilterAndLocs(cycleData[totalCycleCount].tileInfo, locs);
-                            }
-                            cacheTile(totalCycleCount, cycleData[totalCycleCount].tileInfo, currentCycleData);
+        for (final int outputLength : outputLengths) {
+            for (int cycle = 0; cycle < outputLength; cycle++) {
+                final CycleData currentCycleData = cycleData[totalCycleCount];
+                try {
+                    if (cachedTile[totalCycleCount] == null) {
+                        if (!cachedFilter.containsKey(cycleData[totalCycleCount].tileInfo.tileNum)) {
+                            cacheFilterAndLocs(cycleData[totalCycleCount].tileInfo, locs);
                         }
-                    } catch (final IOException e) {
-                        // when logging the error, increment cycle by 1, since totalCycleCount is zero-indexed but Illumina directories are 1-indexed.
-                        throw new PicardException(String.format("Error while reading from BCL file for cycle %d. Offending file on disk is %s",
-                                (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()), e);
+                        cacheTile(totalCycleCount, cycleData[totalCycleCount].tileInfo, currentCycleData);
                     }
-                    totalCycleCount++;
+                } catch (final IOException e) {
+                    // when logging the error, increment cycle by 1, since totalCycleCount is zero-indexed but Illumina directories are 1-indexed.
+                    throw new PicardException(String.format("Error while reading from BCL file for cycle %d. Offending file on disk is %s",
```

Comment:
Should this be CBCL?
```suggestion
                    throw new PicardException(String.format("Error while reading from CBCL file for cycle %d. Offending file on disk is %s",
```